pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -220,17 +238,67 @@ func (c *TargetConfigController) manageRecoveryPod(substitutionReplacer *strings
 }
 
-func (c *TargetConfigController) manageStandardPod(substitutionReplacer *strings.Replacer, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
+func (c *TargetConfigController) manageStandardPod(substitutionReplacer *strings.Replacer, client coreclientv1.ConfigMapsGetter, cluserVersionsClient configv1versionedclient.ClusterVersionsGetter, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
 	podBytes := etcd_assets.MustAsset("etcd/pod.yaml")
 	substitutedPodString := substitutionReplacer.Replace(string(podBytes))
 
+	// Remove the rollbackcopier container if desired.
+	clusterVersion, err := cluserVersionsClient.ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+	pod := &corev1.Pod{}
+	if err := yaml.Unmarshal(podBytes, pod); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal etcd pod: %w", err)
+	}
+	updatedPod := ensureRollbackCopier(pod, clusterVersion)
+	if !equality.Semantic.DeepEqual(pod, updatedPod) {
+		newPodBytes, err := yaml.Marshal(updatedPod)
+		if err != nil {
+			return nil, false, fmt.Errorf("failed to marshal etcd pod: %w", err)
+		}
+		podBytes = newPodBytes
+	}
+
 	podConfigMap := resourceread.ReadConfigMapV1OrDie(etcd_assets.MustAsset("etcd/pod-cm.yaml"))
 	podConfigMap.Data["pod.yaml"] = substitutedPodString
 	podConfigMap.Data["forceRedeploymentReason"] = operatorSpec.ForceRedeploymentReason
 	podConfigMap.Data["version"] = version.Get().String()
 	return resourceapply.ApplyConfigMap(client, recorder, podConfigMap)
 }
 
+func ensureRollbackCopier(etcdPod *corev1.Pod, clusterVersion *configv1.ClusterVersion) *corev1.Pod {
+	errors := []error{}
+	if v, err := semver.Parse(clusterVersion.Status.Desired.Version); err != nil {
+		errors = append(errors, fmt.Errorf("unrecognizable cluster version %q: %w", clusterVersion.Status.Desired.Version, err))
+	} else {
+		if v.Major == 4 && v.Minor <= 5 {
+			errors = append(errors, fmt.Errorf("cluster version %q must be >= 4.5.0", v))
+		}
+	}
+	if clusteroperatorv1helpers.IsStatusConditionFalse(clusterVersion.Status.Conditions, configv1.OperatorAvailable) {
+		errors = append(errors, fmt.Errorf("clusterversion condition %s=False", configv1.OperatorAvailable))
+	}
+	if clusteroperatorv1helpers.IsStatusConditionTrue(clusterVersion.Status.Conditions, configv1.OperatorProgressing) {
+		errors = append(errors, fmt.Errorf("clusterversion condition %s=True", configv1.OperatorProgressing))
+	}
+	if clusteroperatorv1helpers.IsStatusConditionTrue(clusterVersion.Status.Conditions, "Failing") {
+		errors = append(errors, fmt.Errorf("clusterversion condition %s=True", "Failing"))
+	}
+	if err := utilerrors.NewAggregate(errors); err != nil {
+		klog.V(3).Infof("won't remove rollback copier because following preconditions failed: %v", err)
+		return etcdPod
+	}
```

Comment:
Pretty sure this is unworkable without additional state because it'll falsely say the copier should exist during a 4.5.z->4.5.z+1 or (e.g.) 4.5->4.6 upgrade